src/components/Common/Module/FieldPicker.vue

```diff
@@ -1,46 +1,158 @@
 <template>
     <fieldset class="form-group">
+      <b-input-group
+        class="w-25 mt-1 mb-3"
+      >
+        <b-input
+          v-model.trim="query"
+          class="border-right-0 text-truncate"
+          type="search"
+          :placeholder="$t('module.searchPlaceholder')" />
+        <b-input-group-append>
+          <b-input-group-text class="text-primary bg-white border-left-0">
+            <font-awesome-icon
+              :icon="['fas', 'search']"
+            />
+          </b-input-group-text>
+        </b-input-group-append>
+      </b-input-group>
       <div class="fields d-flex">
         <div class="available">
-          <label>{{ $t('field.selector.available') }}</label>
-          <b-button @click.prevent="selectedFields = [...allFields]" variant="link" class="float-right">{{ $t('field.selector.selectAll') }}</b-button>
-          <draggable
-            class="drag-area border"
-            :list.sync="availableFields"
-            :group="group"
+          <label
+            class="text-primary"
           >
-            <div v-for="field in availableFields"
-                 @dblclick="selectedFields = [...selectedFields, field]"
-                 class="field"
-                 :key="field.name">
-              <span v-if="field.label">{{ field.label }} ({{field.name}})</span>
-              <span v-else>{{field.name}}</span>
-              <span v-if="field.isRequired">*</span>
-              <span class="system float-right" v-if="field.isSystem">{{ $t('field.selector.systemField') }}</span>
-            </div>
-          </draggable>
+            {{ $t('field.selector.available') }}
+          </label>
+          <b-button
+            @click.prevent="selectedFields = [...allFields]"
+            variant="link"
+            class="text-muted float-right"
+          >
+            {{ $t('field.selector.selectAll') }}
+          </b-button>
+          <b-list-group
+            vertical
+            class="overflow-auto"
+          >
+            <draggable
+              class="drag-area"
+              :list.sync="availableFields"
+              :group="group"
+            >
+              <b-list-group-item
+                v-for="field in filteredFields(availableFields)"
+                :key="field.name"
+                @dblclick="selectedFields = [...selectedFields, field]"
+                class="selected mb-3 pl-3 border rounded text-truncate"
+              >
+                <font-awesome-icon
+                  class="float-left align-baseline m-1 mr-3 text-primary"
```

Comment:
Add grip class here (pointer: grip).
Also make sure only this part of the item is actually draggable.
Look for an example elsewhere in the repo. Hint: the prop is called `handle` and it references a class that can be dragged